src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -358,7 +358,7 @@ private int FindEntry(TKey key)
                 int hashCode = (comparer != null ? comparer.GetHashCode(key) : key.GetHashCode()) & 0x7FFFFFFF;
                 for (int i = buckets[hashCode % buckets.Length]; i >= 0; i = entries[i].next)
                 {
-                    if (entries[i].hashCode == hashCode && (comparer != null ? comparer.Equals(entries[i].key, key) : (default(TKey) == null ? key.Equals(entries[i].key) : EqualityComparer<TKey>.Default.Equals(entries[i].key, key)))) return i;
+                    if (entries[i].hashCode == hashCode && (comparer != null ? comparer.Equals(entries[i].key, key) : EqualityComparer<TKey>.Default.Equals(entries[i].key, key))) return i;
```

Comment:
why without `ref Entry`?

Reply:
Going to follow up with ref changes

Reply:
e.g. just pulling in the devirtualization of `EqualityComparer<T>.Default`  https://github.com/dotnet/coreclr/pull/14125 to skip the interface dispatch